compiler/noirc_frontend/src/hir/def_collector/dc_mod.rs

```diff
@@ -361,7 +377,25 @@ impl<'a> ModCollector<'a> {
                 module_id: self.module_id,
                 type_alias_def: type_alias,
             };
-            let resolved_generics = context.resolve_generics(&unresolved.type_alias_def.generics);
+
+            let generic_results = context.resolve_generics(&unresolved.type_alias_def.generics);
+            let mut resolved_generics = Vec::new();
+            for generic_result in generic_results.into_iter() {
+                match generic_result {
+                    Ok(resolved_generic) => resolved_generics.push(resolved_generic),
+                    Err((bad_generic, bad_type)) => {
+                        let err = DefCollectorErrorKind::UnsupportedNumericGenericType {
+                            ident: Ident::new(bad_generic.name.to_string(), bad_generic.span),
+                            typ: bad_type.clone(),
+                        };
+                        errors.push((err.into(), self.file_id));
+                        // We still push the ill-formed generic as to continue definition collection and elaboration
+                        // without triggering any other compiler errors.
+                        resolved_generics.push(bad_generic);
+                    }
+                }
+            }
```

Comment:
Yeah, I think these unwieldy, copied loops are the consequence of the return type chosen for `resolve_generics` - we should change it.

Reply:
Yup much cleaner when we directly transform the error and with the improved return type. Just pushed the change.